### ISSUE TITLE
Indentation in code chunks

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [rstudio/learnr#116]
It would be useful to automatically ident code like the RStudio editor does. For example

```
ggplot(diamonds, aes(x = color)) +
geom_bar()
```

should be

```
ggplot(diamonds, aes(x = color)) +
  geom_bar()
```

by default, which would require that hitting Enter after `+` in the first line takes you to the indented spot.

**Comment by jjallaire** [comment]
Yes, we would indeed like to bring some of our automatic indentation code over from RStudio. @kevinushey What are the obstacles to this? (I believe that it's partly because it's wrapped up with our document model which is harder to extract)

**Comment by kevinushey** [comment]
That's right -- the RStudio indentation code makes use of some of our own custom R code model + tokenization code (alongside other JavaScript utilities), so we have to be a bit more careful in trying to port those over. It's should certainly be doable, though.

**Comment by krlmlr** [comment]
But we could adopt *styler*? CC @lorenzwalthert.

**Comment by mine-cetinkaya-rundel** [comment]
That would be awesome @krlmlr. I was actually looking into running the code through styler in the checking stage and providing feedback. I haven't really figured it out yet, but I'd love for it to be a feature of learnr.

**Comment by krlmlr** [comment]
I haven't read closely enough, not sure how useful automatic styling would be. (Perhaps a button that styles the code when clicked?)

For adding indention after pressing Enter, a simple hack may be to look at the last symbol / character in the line and at parentheses mismatch: if the line ends with an operator or there are more opening than closing parens/braces, we indent by two more spaces. Perhaps this gets many cases right already?
